mmf/datasets/base_dataset.py

```diff
@@ -24,7 +25,11 @@ def __init__(self, dataset_name, config, dataset_type="train", *args, **kwargs):
         self._dataset_name = dataset_name
         self._dataset_type = dataset_type
         self._global_config = registry.get("config")
-        self._device = registry.get("current_device")
+        self._device = (
+            f"cuda:{torch.cuda.current_device()}"
+            if torch.cuda.is_available()
+            else torch.device("cpu")
+        )
```

Comment:
This can be made into an util function, since I see this being used in multiple places.